lib/_emerge/Scheduler.py

```diff
@@ -1792,6 +1801,47 @@ def _is_work_scheduled(self):
     def _running_job_count(self):
         return self._jobs
 
+    def _can_add_job(self):
+        if not super()._can_add_job():
+            return False
+
```

Comment:
I think this requires a check that returns true if there are no running jobs, as otherwise we will never add a job but just stall indefinitely.

Reply:
I agree that it's a useful property to always allow at least one job, however there could be an edge case where this allows a job to start at an inappropriate time, like just before it's supposed to start processing packages from _merge_wait_queue which only happens when the running job count is zero.

I think it will probably be fine because _schedule_tasks processes _merge_wait_queue before it calls _schedule_tasks_imp to start new jobs, but we need to be careful and should add a comment about this scheduling order requirement in _schedule_tasks.

Reply:
> I agree that it's a useful property

I'ts not only a useful property, something like this is required to prevent portage from hanging if the tmpdir space requirement is not fulfilled. Try it for yourself

```
$ git checkout 0f2a0b2b5a8456821747a698cd98fb15228fd37f # This PRs commit
$ sudo ./bin/emerge --jobs-tmpdir-blocks-threshold 0.01 --jobs 3 -v app-misc/hello app-text/pdfjam 
# hangs
```

I currently use

https://github.com/gentoo/portage/pull/1353/files#diff-e07339ce877e9437fcd80afb3da63e264d5ae26a50dd27d6bab58a6019ec1b73R1815-R1819

however, we could alternatively bail with an error if the tmpdir resource requirements are not meet and there _merge_wait_queue is empty.

Reply:
Okay, I've copied your code:
```python
        running_job_count = self._running_job_count()
        if running_job_count == 0 and not self._merge_wait_queue:
            # Ensure there is forward progress if there are no running
            # jobs and no jobs in the _merge_wait_queue.
            return True
```